lib/chef/node/immutable_collections.rb

```diff
@@ -44,7 +44,7 @@ def convert_value(value)
         when Array
           ImmutableArray.new(value, __root__, __node__, __precedence__)
         else
-          safe_dup(value).freeze
+          value.frozen? ? value : safe_dup(value).freeze
```

Comment:
Pondering the case in which there's a way to break this check:
* `BasicObject` subclasses
* Native extensions?
* `defrost` gem

Reply:
From meeting: we'll do this only for strings.